### PR TITLE
Add support for Node Info

### PR DESCRIPTION
**Pull request by chen-keinan**
Add support for triggering `node-collector` job in cluster.
the collector will collect info from node and output it.
require to support both for trivy and operator.
Related :

- [epic](https://github.com/aquasecurity/trivy/issues/2198)
-  [story](https://github.com/aquasecurity/trivy/issues/2201)

### CODE REVIEW

File: examples/trivy.go [hunk 1]
```diff
@@ -36,41 +36,53 @@ func main() {
 		log.Fatal(err)
 	}
 	printArtifacts(artifacts)
+	/*
```
Comment (josedonizetti) [inline examples/trivy.go:39]: remove the commented code, or add back if necessary?

File: pkg/jobs/collector.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		WithJobServiceAccount(jb.serviceAccount),
+		WithJobName(fmt.Sprintf("%s-%s", jb.templateName, nodeName)))
+	if err != nil {
+		return "", fmt.Errorf("running node-collector job: %w", err)
+	}
+
+	_, err = jb.getTrivyNamespace(ctx)
+	if err != nil {
+		if k8sapierror.IsNotFound(err) {
+			trivyNamespace := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: TrivyNamespace}}
```
Comment (josedonizetti) [inline pkg/jobs/collector.go:115]: Why do we create the namespace for `ApplyAndCollect`, but not for `Apply`? `Apply` also creates a job.

File: pkg/trivyk8s/trivyk8s.go [hunk 3]
```diff
@@ -105,6 +110,35 @@ func (c *client) ListArtifacts(ctx context.Context) ([]*artifacts.Artifact, erro
...
 
+// ListArtifacts returns kubernetes scannable artifacs.
+func (c *client) ListArtifactAndNodeInfo(ctx context.Context) ([]*artifacts.Artifact, error) {
+	artifactList, err := c.ListArtifacts(ctx)
+	if err != nil {
+		return nil, err
+	}
+	jc := jobs.NewCollector(c.cluster, jobs.WithTimetout(time.Minute*5), jobs.WithJobTemplateName(jobs.NodeCollectorName))
+	// delete trivy namespace
+	defer jc.Cleanup(ctx)
```
Comment (josedonizetti) [inline pkg/trivyk8s/trivyk8s.go:121]: We cannot clean the namespace if it existed before, or we need a new namespace. Running the example here to get the nodes info deleted my trivy-operator `trivy-system` namespace!

**Review by josedonizetti — CHANGES_REQUESTED** [review]
@chen-keinan The deletion of the namespace was a big deal, the reason why I'm blocking the pr.

File: pkg/jobs/collector.go (hunk 2 above)
Comment (chen-keinan) [reply pkg/jobs/collector.go:115]: `ApplyAndCollect` is for trivy and  `Apply` is for operator , so no need to create namespace for operator

File: pkg/jobs/template/node-collector.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: node-collector
+spec:
+  template:
+    metadata:
```
Comment (josedonizetti) [inline pkg/jobs/template/node-collector.yaml:8]: can we add more specific labels here? To show that this image is created by trivy, etc.

**Comment by chen-keinan** [comment]
@josedonizetti all comments has been addressed , I'm merging the PR to be able to continue with other dependents PR.

**Pull request merged**